lib/src/rules/unsafe_unicode.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../util/unicode_utils.dart';
+
+const _desc = r"Don't use bidirectional Unicode text.";
+
+const _details = r'''
+
+**AVOID** using bidirectional Unicode text.
+
+[Bidirectional Unicode](https://unicode.org/reports/tr9/) text may be
+interpreted and compiled differently than how it appears in editors
+leading to possible security vulnerabilities. 
+
```

Comment:
Do we also want to say something like:

```
For rare cases where bidirectional unicode text is needed,
prefix the statement with `// ignore: unsafe_uncode` and a
comment explaining that the intended use is.
```

Reply:
Updated.

Reply:
The implementation does not support this usage, so either the comment or the implementation should change.